src/Libraries/CoreNodeModels/Input/BaseTypes.cs

```diff
@@ -103,12 +103,29 @@ public DoubleInput()
 
             ShouldDisplayPreviewCore = false;
             ConvertToken = Convert;
-            Value = "0";
+            NumericalValue = "0";
 
             //ws.DynamoModel.PreferenceSettings.PropertyChanged += Preferences_PropertyChanged;
         }
 
-        public virtual double Convert(double value)
+        //if the string can be parsed to a common numeric type return true
+        internal bool validateInput(string value)
```

Comment:
@mjkkirschner Sorry I'm late to looking at this, but I think this logic is living in the wrong place. Input validation should live in the View layer. This should be handled via a converter applied to the binding on the `Value` property. Interestingly, your `Convert(..)` and `CanConvert(...)` methods are exactly the same as in the `IValueConverter` interface where we use in every other case where we need to do input validation. Why reinvent the wheel? This will also allow us to avoid having a property called `NumericalValue` which stores a string, which is a bit confusing).